xwiki-platform-core/xwiki-platform-refactoring/xwiki-platform-refactoring-api/src/main/java/org/xwiki/refactoring/internal/job/AbstractCopyOrMoveJob.java

```diff
@@ -309,14 +307,28 @@ protected void process(final SpaceReference source, final SpaceReference destina
 
     protected boolean checkAllRights(DocumentReference oldReference, DocumentReference newReference) throws Exception
     {
-        if (!hasAccess(Right.VIEW, oldReference)) {
-            this.logger.error("You don't have sufficient permissions over the source document [{}].", oldReference);
-            return false;
-        } else if (!hasAccess(Right.EDIT, newReference)
-            || (this.modelBridge.exists(newReference) && !hasAccess(Right.DELETE, newReference))) {
-            this.logger.error("You don't have sufficient permissions over the destination document [{}].",
-                newReference);
-            return false;
+        boolean isWebPreferences =
+            isSpacePreferencesReference(oldReference) && isSpacePreferencesReference(newReference);
+        if (!isWebPreferences) {
+            if (!hasAccess(Right.VIEW, oldReference)) {
+                this.logger.error("You don't have sufficient permissions over the source document [{}].", oldReference);
+                return false;
+            } else if (!hasAccess(Right.EDIT, newReference)
+                || (this.modelBridge.exists(newReference) && !hasAccess(Right.DELETE, newReference)))
+            {
+                this.logger.error("You don't have sufficient permissions over the destination document [{}].",
+                    newReference);
+                return false;
+            }
+        } else {
+            DocumentReference newHomeReference = getSpaceHomeReference(newReference);
+            DocumentReference oldHomeReference = getSpaceHomeReference(oldReference);
+            if (!this.concernedEntities.containsKey(oldHomeReference) || !hasAccess(Right.EDIT, newHomeReference)) {
+                this.logger.error("You don't have sufficient permissions over the home document of WebPreferences "
```

Comment:
This message seems misleading for the first case where the entities to be moved don't contain the main page of the space. Also, thinking of https://forum.xwiki.org/t/batch-move-pages-design/16559, can't we imagine a move operation that moves all child pages but not the main page and where the pinning state of these child pages should be moved?

Reply:
> This message seems misleading for the first case where the entities to be moved don't contain the main page of the space

Right

> Also, thinking of https://forum.xwiki.org/t/batch-move-pages-design/16559, can't we imagine a move operation that moves all child pages but not the main page and where the pinning state of these child pages should be moved?

Apparently this wasn't discussed in https://forum.xwiki.org/t/copy-move-delete-the-webpreferences-page-along-with-its-parent-page/16332 so I'm not really sure how we would deal on such scenario. I'm wondering if it wouldn't be an actual option to add in the batch move new design. In any case I won't address that use case here: we'd create an improvment for it.

Reply:
> This message seems misleading for the first case where the entities to be moved don't contain the main page of the space

Actually the condition was wrong, I fixed it.